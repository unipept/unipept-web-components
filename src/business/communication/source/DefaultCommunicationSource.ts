import CommunicationSource from "./CommunicationSource";
import EcResponseCommunicator from "@/business/communication/functional/ec/EcResponseCommunicator";
import GoResponseCommunicator from "./../functional/go/GoResponseCommunicator";
import InterproResponseCommunicator from "./../functional/interpro/InterproResponseCommunicator";
import NcbiResponseCommunicator from "./../taxonomic/ncbi/NcbiResponseCommunicator";
import Pept2DataCommunicator from "./../peptides/Pept2DataCommunicator";

export default class DefaultCommunicationSource implements CommunicationSource {
    public constructor(
        private readonly serviceUrl: string
    ) {}

    public getEcCommunicator(): EcResponseCommunicator {
        return new EcResponseCommunicator();
    }

    public getGoCommunicator(): GoResponseCommunicator {
        return new GoResponseCommunicator();
    }

    public getInterproCommunicator(): InterproResponseCommunicator {
        return new InterproResponseCommunicator();
    }

    public getNcbiCommunicator(): NcbiResponseCommunicator {
        return new NcbiResponseCommunicator();
    }

    public getPept2DataCommunicator(): Pept2DataCommunicator {
        return new Pept2DataCommunicator(this.serviceUrl);
    }
}
