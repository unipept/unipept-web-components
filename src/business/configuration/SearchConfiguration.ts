import Visitable from "../visitor/Visitable";
import SearchConfigurationVisitor from "./SearchConfigurationVisitor";

export default class SearchConfiguration implements Visitable<SearchConfigurationVisitor> {
    constructor(
        public equateIl: boolean = true,
        public filterDuplicates: boolean = true,
        public enableMissingCleavageHandling: boolean = false,
        public id?: string
    ) {}

    public toString() {
        return [this.equateIl, this.filterDuplicates, this.enableMissingCleavageHandling].map(
            t => t.toString()
        ).join(",");
    }

    public async accept(visitor: SearchConfigurationVisitor): Promise<void> {
        visitor.visitSearchConfiguration(this);
    }
}

