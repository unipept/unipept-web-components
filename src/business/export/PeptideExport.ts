import { CountTable } from "./../counts/CountTable";
import { Peptide } from "./../ontology/raw/Peptide";
import { NcbiRank } from "./../ontology/taxonomic/ncbi/NcbiRank";
import { GoNamespace } from "./..//ontology/functional/go/GoNamespace";
import Pept2DataCommunicator from "./../communication/peptides/Pept2DataCommunicator";
import SearchConfiguration from "./../configuration/SearchConfiguration";
import NcbiTaxon, { NcbiId } from "./../ontology/taxonomic/ncbi/NcbiTaxon";
import { Ontology, OntologyIdType } from "./../ontology/Ontology";
import LcaCountTableProcessor from "./../processors/taxonomic/ncbi/LcaCountTableProcessor";
import NcbiOntologyProcessor from "./../ontology/taxonomic/ncbi/NcbiOntologyProcessor";
import GoCountTableProcessor from "./../processors/functional/go/GoCountTableProcessor";
import GoOntologyProcessor from "./../ontology/functional/go/GoOntologyProcessor";
import EcCountTableProcessor from "./../processors/functional/ec/EcCountTableProcessor";
import FunctionalCountTableProcessor from "./../processors/functional/FunctionalCountTableProcessor";
import FunctionalDefinition from "./../ontology/functional/FunctionalDefinition";
import OntologyProcessor from "./../ontology/OntologyProcessor";
import EcOntologyProcessor from "./../ontology/functional/ec/EcOntologyProcessor";
import InterproCountTableProcessor from "./../processors/functional/interpro/InterproCountTableProcessor";
import InterproOntologyProcessor from "./../ontology/functional/interpro/InterproOntologyProcessor";
import StringUtils from "./../misc/StringUtils";
import { PeptideDataResponse } from "./../communication/peptides/PeptideDataResponse";
import EcDefinition from "./../ontology/functional/ec/EcDefinition";
import GoDefinition from "./../ontology/functional/go/GoDefinition";
import InterproDefinition from "./../ontology/functional/interpro/InterproDefinition";
import CommunicationSource from "./../communication/source/CommunicationSource";
import { NcbiResponseCommunicator, PeptideData } from "@/business";
import { ShareableMap } from "shared-memory-datastructures";

export default class PeptideExport {
    /**
     * Produces a CSV that contains one row per peptide (these rows are duplicated for peptides that occur multiple
     * times). Every row contains the LCA for a peptide, as well as it's associated lineage, and the top 3 most
     * occurring annotations for the EC-numbers, the three different GO-domains and InterPro-annotations.
     *
     * @param peptideCountTable Count table that contains all peptides and their associated counts that should be
     * present in the generated export.
     * @param searchConfiguration The particular configuration settings that are used for processing the peptides
     * present in the count table.
     * @param pept2data
     * @param separator The delimiter used to separate columns in the CSV. Use comma for international format, and semi-
     * colon for the European version.
     * @param secondarySeparator The delimiter used to separate multiple functional annotations from each other. Some
     * columns (like the EC-number list) contain multiple items, which need to be separated from each other using a
     * character different from the default separator.
     * @param lineEnding The line terminator that should be used.
     */
    public static async exportSummaryAsCsv(
        peptideCountTable: CountTable<Peptide>,
        searchConfiguration: SearchConfiguration,
        pept2data: ShareableMap<Peptide, PeptideData>,
        communicationSource: CommunicationSource,
        separator: string = ",",
        secondarySeparator: string = ";",
        lineEnding: string = "\n"
    ): Promise<string> {
        const rows: string[] = [];
        rows.push(PeptideExport.getHeader().join(separator));

        const ncbiOntology = await PeptideExport.computeNcbiOntology(
            peptideCountTable,
            searchConfiguration,
            pept2data,
            communicationSource.getNcbiCommunicator()
        );
        const goOntology = await PeptideExport.computeFunctionalOntology(
            new GoCountTableProcessor(peptideCountTable, searchConfiguration, pept2data, communicationSource.getGoCommunicator(), 0),
            new GoOntologyProcessor(communicationSource.getGoCommunicator())
        );
        const ecOntology = await PeptideExport.computeFunctionalOntology(
            new EcCountTableProcessor(peptideCountTable, searchConfiguration, pept2data, communicationSource.getEcCommunicator(), 0),
            new EcOntologyProcessor(communicationSource.getEcCommunicator())
        );
        const interproOntology = await PeptideExport.computeFunctionalOntology(
            new InterproCountTableProcessor(peptideCountTable, searchConfiguration, pept2data, communicationSource.getInterproCommunicator(), 0),
            new InterproOntologyProcessor(communicationSource.getInterproCommunicator())
        );

        const headerLength = PeptideExport.getHeader().length;

        for (const peptide of peptideCountTable.getOntologyIds()) {
            const row = [];

            row.push(peptide);

            const pept2DataResponse = pept2data.get(peptide);

            if (!pept2DataResponse) {
                for (let i = 0; i < headerLength - 1; i++) {
                    row.push("");
                }
            } else {
                // First construct the taxonomical part of each output row.
                const lcaDefinition = ncbiOntology.getDefinition(pept2DataResponse.lca);
                row.push(lcaDefinition ? lcaDefinition.name : "");

                const processedLineage = pept2DataResponse.lineage.map(l => l ? ncbiOntology.getDefinition(l) : null);
                row.push(...processedLineage.map(l => l ? l.name : ""));

                // Now add information about the EC-numbers.
                // This list contains the (EC-code, protein count)-mapping, sorted descending on counts.
                const ecNumbers = await PeptideExport.sortAnnotations(pept2DataResponse.ec);
                row.push(
                    ecNumbers
                        .map(a => `${a[0].substr(3)} (${StringUtils.numberToPercent(a[1] / pept2DataResponse.faCounts.ec)})`)
                        .join(secondarySeparator)
                );

                const ecDefinitions: [EcDefinition, number][] = ecNumbers.map(c => [ecOntology.getDefinition(c[0]), c[1]]);
                row.push(
                    ecDefinitions.map(
                        c => `${c[0] ? c[0].name : ""} (${StringUtils.numberToPercent(c[1] / pept2DataResponse.faCounts.ec)})`
                    ).join(secondarySeparator)
                );

                // Now process the GO-terms
                for (const ns of Object.values(GoNamespace)) {
                    const gos = pept2DataResponse.go;
                    const goAnnotations = Object.keys(gos).filter(
                        x => goOntology.getDefinition(x).namespace === ns
                    );

                    const goTerms = {};

                    for (const annotation of goAnnotations) {
                        goTerms[annotation] = gos[annotation];
                    }

                    const sortedTerms = PeptideExport.sortAnnotations(goTerms);

                    row.push(
                        sortedTerms
                            .map(a => `${a[0]} (${StringUtils.numberToPercent(a[1] / pept2DataResponse.faCounts.go)})`)
                            .join(secondarySeparator)
                    );

                    const goDefinitions: [GoDefinition, number][] = sortedTerms.map(c => [goOntology.getDefinition(c[0]), c[1]]);
                    row.push(
                        goDefinitions.map(
                            c => `${c ? c[0].name : ""} (${StringUtils.numberToPercent(c[1] / pept2DataResponse.faCounts.go)})`
                        ).join(secondarySeparator));
                }

                // Now process the InterPro-terms
                const interproNumbers = await PeptideExport.sortAnnotations(pept2DataResponse.ipr);
                row.push(
                    interproNumbers
                        .map(a => `${a[0].substr(4)} (${StringUtils.numberToPercent(a[1] / pept2DataResponse.faCounts.ipr)})`)
                        .join(secondarySeparator)
                );

                const interproDefinitions: [InterproDefinition, number][] = interproNumbers.map(i => [interproOntology.getDefinition(i[0]), i[1]]);
                row.push(interproDefinitions.map(
                    i => `${i && i[0] ? i[0].name : ""} (${StringUtils.numberToPercent(i[1] / pept2DataResponse.faCounts.ipr)})`
                ).join(secondarySeparator));
            }

            const rowString = row.join(separator);
            // Duplicate row in case that the peptide is present more than once.
            for (let i = 0; i < peptideCountTable.getCounts(peptide); i++) {
                rows.push(rowString);
            }
        }

        return rows.join(lineEnding);
    }

    private static sortAnnotations(
        annotations: any
    ): [string, number][] {
        return (Object.entries(annotations) as [string, any][] as [string, number][])
            .sort((a, b) => {
                if (b[1] === a[1]) {
                    return a[0] < b[0] ? -1 : 1;
                }
                return b[1] - a[1]
            })
            .slice(0, 3);
    }

    /**
     * @return The default set of columns that's part of the generated export.
     */
    private static getHeader(): string[] {
        const headers: string[] = [];
        headers.push("peptide");
        headers.push("lca");
        headers.push(...Object.values(NcbiRank));
        headers.push("EC");
        headers.push("EC - names")
        headers.push(...Object.values(GoNamespace).map(ns => `GO (${ns})`));
        headers.push(...Object.values(GoNamespace).map(ns => `GO (${ns}) - names`));
        headers.push("InterPro");
        headers.push("InterPro - names");
        return headers;
    }

    /**
     * @param peptideCountTable The count table with all peptides for whom associated peptide counts must be looked up.
     * @param searchConfiguration Search settings needed to process the peptide count table.
     * @return An ontology that contains information about all NCBI-taxa that are associated with the given peptide
     * count table.
     */
    private static async computeNcbiOntology(
        peptideCountTable: CountTable<Peptide>,
        searchConfiguration: SearchConfiguration,
        pept2Data: ShareableMap<Peptide, PeptideData>,
        ncbiCommunicator: NcbiResponseCommunicator
    ): Promise<Ontology<NcbiId, NcbiTaxon>> {
        const taxaProcessor = new LcaCountTableProcessor(peptideCountTable, searchConfiguration, pept2Data);
        const lcaCountTable = await taxaProcessor.getCountTable();

        const ncbiOntologyProcessor = new NcbiOntologyProcessor(ncbiCommunicator);
        return await ncbiOntologyProcessor.getOntology(lcaCountTable);
    }

    private static async computeFunctionalOntology<OntologyId extends OntologyIdType, OntologyDefinition extends FunctionalDefinition>(
        countProcessor: FunctionalCountTableProcessor<OntologyId, OntologyDefinition>,
        ontologyProcessor: OntologyProcessor<OntologyId, OntologyDefinition>
    ): Promise<Ontology<OntologyId, OntologyDefinition>> {
        const countTable = await countProcessor.getCountTable();
        return await ontologyProcessor.getOntology(countTable);
    }
}
