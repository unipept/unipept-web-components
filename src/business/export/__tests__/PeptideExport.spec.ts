import Mock from "@/test/Mock";
import PeptideExport from "@/business/export/PeptideExport";
import SearchConfiguration from "@/business/configuration/SearchConfiguration";
import DefaultCommunicationSource from "@/business/communication/source/DefaultCommunicationSource";
import Setup from "@/test/Setup";

describe("PeptideExport", () => {
    beforeEach(() => {
        const setup = new Setup();
        setup.setupAll();
    })

    it("produces a correct CSV output", async(done) => {
        // const mock = new Mock();
        // const peptideCountTable = await mock.mockRealisticPeptideCountTable();
        //
        // const result = await PeptideExport.exportSummaryAsCsv(
        //     peptideCountTable,
        //     new SearchConfiguration(),
        //     new DefaultCommunicationSource()
        // );
        // expect(result).toMatchSnapshot();
        //
        // done();
    });
});
