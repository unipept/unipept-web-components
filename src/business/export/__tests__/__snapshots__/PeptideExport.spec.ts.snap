// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeptideExport produces a correct CSV output 1`] = `
"peptide,lca,superkingdom,kingdom,subkingdom,superphylum,phylum,subphylum,superclass,class,subclass,infraclass,superorder,order,suborder,infraorder,parvorder,superfamily,family,subfamily,tribe,subtribe,genus,subgenus,species group,species subgroup,species,subspecies,varietas,forma,EC,EC - names,GO (cellular component),GO (molecular function),GO (biological process),GO (cellular component) - names,GO (molecular function) - names,GO (biological process) - names,InterPro,InterPro - names
YVVLQPGVK,Bacteroides,Bacteria,,,,Bacteroidetes,,,Bacteroidia,,,,Bacteroidales,,,,,Bacteroidaceae,,,,Bacteroides,,,,,,,,2.7.7.6 (200%),DNA-directed RNA polymerase (200%),,,GO:0000287 (100%);GO:0003677 (100%);GO:0003899 (100%),magnesium ion binding (100%);DNA binding (100%);DNA-directed 5'-3' RNA polymerase activity (100%),GO:0006351 (100%),transcription, DNA-templated (100%),IPR000722 (100%);IPR006592 (100%);IPR007066 (100%),RNA polymerase, alpha subunit (100%);RNA polymerase, N-terminal (100%);RNA polymerase Rpb1, domain 3 (100%)
SGLAPFYSDK,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
FLGFEQLFK,root,,,,,,,,,,,,,,,,,,,,,,,,,,,,,1.4.1.4 (200%),Glutamate dehydrogenase (NADP(+)) (200%),GO:0005829 (38%);GO:0005737 (31%);GO:0005634 (19%),cytosol (38%);cytoplasm (31%);nucleus (19%),GO:0004354 (100%),glutamate dehydrogenase (NADP+) activity (100%),GO:0006537 (56%);GO:0006520 (44%);GO:0019676 (13%),glutamate biosynthetic process (56%);cellular amino acid metabolic process (44%);ammonia assimilation cycle (13%),IPR006095 (100%);IPR006096 (100%);IPR006097 (100%),Glutamate/phenylalanine/leucine/valine dehydrogenase (100%);Glutamate/phenylalanine/leucine/valine dehydrogenase, C-terminal (100%);Glutamate/phenylalanine/leucine/valine dehydrogenase, dimerisation domain (100%)
LVLVNAVYFR,Hominidae,Eukaryota,Metazoa,,,Chordata,Craniata,Sarcopterygii,Mammalia,,,Euarchontoglires,Primates,Haplorrhini,Simiiformes,Catarrhini,Hominoidea,Hominidae,,,,,,,,,,,,,,GO:0005615 (100%);GO:0005737 (100%);GO:0005576 (50%),extracellular space (100%);cytoplasm (100%);extracellular region (50%),GO:0004867 (100%);GO:0002020 (50%),serine-type endopeptidase inhibitor activity (100%);protease binding (50%),GO:0007605 (50%);GO:0010951 (50%);GO:0043312 (50%),sensory perception of sound (50%);negative regulation of endopeptidase activity (50%);neutrophil degranulation (50%),IPR000215 (100%);IPR023795 (100%);IPR023796 (100%),Serpin family (100%);Serpin, conserved site (100%);Serpin domain (100%)
FATSDLNDLYR,root,,,,,,,,,,,,,,,,,,,,,,,,,,,,,2.7.7.6 (200%),DNA-directed RNA polymerase (200%),GO:0009507 (1%);GO:0005618 (0%);GO:0005737 (0%),chloroplast (1%);cell wall (0%);cytoplasm (0%),GO:0003677 (100%);GO:0003899 (100%);GO:0000287 (99%),DNA binding (100%);DNA-directed 5'-3' RNA polymerase activity (100%);magnesium ion binding (99%),GO:0006351 (100%);GO:0046677 (0%),transcription, DNA-templated (100%);response to antibiotic (0%),IPR000722 (100%);IPR006592 (100%);IPR007066 (100%),RNA polymerase, alpha subunit (100%);RNA polymerase, N-terminal (100%);RNA polymerase Rpb1, domain 3 (100%)
AAANESFGYNEDELVSSDLVGMR,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
LGENNAELNALAK,Homo sapiens,Eukaryota,Metazoa,,,Chordata,Craniata,Sarcopterygii,Mammalia,,,Euarchontoglires,Primates,Haplorrhini,Simiiformes,Catarrhini,Hominoidea,Hominidae,Homininae,,,Homo,,,,Homo sapiens,,,,3.4.17.2 (200%),Carboxypeptidase B (200%),GO:0005615 (100%),extracellular space (100%),GO:0004180 (100%);GO:0004181 (100%);GO:0008270 (100%),carboxypeptidase activity (100%);metallocarboxypeptidase activity (100%);zinc ion binding (100%),GO:0006508 (100%),proteolysis (100%),IPR000834 (100%);IPR003146 (100%);IPR036990 (100%),Peptidase M14, carboxypeptidase A (100%);Carboxypeptidase, activation peptide (100%);Metallocarboxypeptidase-like, propeptide (100%)
LAEEVLR,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
YEVGTMLELPR,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
GMFSMMNYYLPLK,Clostridiales,Bacteria,,,,Firmicutes,,,Clostridia,,,,Clostridiales,,,,,,,,,,,,,,,,,4.1.1.49 (200%),Phosphoenolpyruvate carboxykinase (ATP) (200%),GO:0005737 (100%),cytoplasm (100%),GO:0004612 (100%);GO:0005524 (100%);GO:0046872 (100%),phosphoenolpyruvate carboxykinase (ATP) activity (100%);ATP binding (100%);metal ion binding (100%),GO:0006094 (100%),gluconeogenesis (100%),IPR001272 (100%);IPR008210 (100%);IPR013035 (100%),Phosphoenolpyruvate carboxykinase, ATP-utilising (100%);Phosphoenolpyruvate carboxykinase, N-terminal (100%);Phosphoenolpyruvate carboxykinase, C-terminal (100%)
TNTLLQSAFFK,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
ANFEGECSEVGMYLAMAR,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
MEVAVGDK,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
VLYSK,root,,,,,,,,,,,,,,,,,,,,,,,,,,,,,2.7.8.7 (113%);2.5.1.90 (38%);3.2.2.23 (13%),Holo-[acyl-carrier-protein] synthase (113%);All-trans-octaprenyl-diphosphate synthase (38%);DNA-formamidopyrimidine glycosylase (13%),GO:0005737 (81%);GO:0005634 (6%);GO:0005886 (5%),cytoplasm (81%);nucleus (6%);plasma membrane (5%),GO:0005524 (69%);GO:0000287 (11%);GO:0008897 (11%),ATP binding (69%);magnesium ion binding (11%);holo-[acyl-carrier-protein] synthase activity (11%),GO:0006457 (66%);GO:0006633 (11%);GO:0000122 (5%),protein folding (66%);fatty acid biosynthetic process (11%);negative regulation of transcription by RNA polymerase II (5%),IPR011032 (69%);IPR020818 (69%);IPR037124 (69%),GroES-like superfamily (69%);GroES chaperonin family (69%);GroES chaperonin superfamily (69%)
QLLVPLLPSLVDR,Homo sapiens,Eukaryota,Metazoa,,,Chordata,Craniata,Sarcopterygii,Mammalia,,,Euarchontoglires,Primates,Haplorrhini,Simiiformes,Catarrhini,Hominoidea,Hominidae,Homininae,,,Homo,,,,Homo sapiens,,,,3.5.1.23 (200%);3.5.1.- (100%),Ceramidase (200%);In linear amides (100%),GO:0000139 (100%);GO:0005576 (100%);GO:0005739 (100%),Golgi membrane (100%);extracellular region (100%);mitochondrion (100%),GO:0005509 (100%);GO:0008270 (100%);GO:0017040 (100%),calcium ion binding (100%);zinc ion binding (100%);N-acylsphingosine amidohydrolase activity (100%),GO:0006670 (100%);GO:0006672 (100%);GO:0006915 (100%),sphingosine metabolic process (100%);ceramide metabolic process (100%);apoptotic process (100%),IPR006823 (100%);IPR031329 (100%);IPR031331 (100%),Neutral/alkaline nonlysosomal ceramidase (100%);Neutral/alkaline non-lysosomal ceramidase, N-terminal (100%);Neutral/alkaline non-lysosomal ceramidase, C-terminal (100%)
ELASLHGTK,Homo sapiens,Eukaryota,Metazoa,,,Chordata,Craniata,Sarcopterygii,Mammalia,,,Euarchontoglires,Primates,Haplorrhini,Simiiformes,Catarrhini,Hominoidea,Hominidae,Homininae,,,Homo,,,,Homo sapiens,,,,3.4.17.2 (200%),Carboxypeptidase B (200%),GO:0005615 (100%),extracellular space (100%),GO:0004180 (100%);GO:0004181 (100%);GO:0008270 (100%),carboxypeptidase activity (100%);metallocarboxypeptidase activity (100%);zinc ion binding (100%),GO:0006508 (100%),proteolysis (100%),IPR000834 (100%);IPR003146 (100%);IPR036990 (100%),Peptidase M14, carboxypeptidase A (100%);Carboxypeptidase, activation peptide (100%);Metallocarboxypeptidase-like, propeptide (100%)
LQTNGAVPDVLQQGR,Homo sapiens,Eukaryota,Metazoa,,,Chordata,Craniata,Sarcopterygii,Mammalia,,,Euarchontoglires,Primates,Haplorrhini,Simiiformes,Catarrhini,Hominoidea,Hominidae,Homininae,,,Homo,,,,Homo sapiens,,,,3.4.21.71 (200%),Pancreatic elastase II (200%),GO:0005576 (100%);GO:0005615 (100%);GO:0005829 (100%),extracellular region (100%);extracellular space (100%);cytosol (100%),GO:0004252 (100%);GO:0017171 (100%),serine-type endopeptidase activity (100%);serine hydrolase activity (100%),GO:0006508 (100%);GO:0070268 (100%),proteolysis (100%);cornification (100%),IPR001254 (100%);IPR001314 (100%);IPR009003 (100%),Serine proteases, trypsin domain (100%);Peptidase S1A, chymotrypsin family (100%);Peptidase S1, PA clan (100%)
SFALNFK,Homo sapiens,Eukaryota,Metazoa,,,Chordata,Craniata,Sarcopterygii,Mammalia,,,Euarchontoglires,Primates,Haplorrhini,Simiiformes,Catarrhini,Hominoidea,Hominidae,Homininae,,,Homo,,,,Homo sapiens,,,,,,GO:0005615 (100%);GO:0005829 (100%);GO:0005886 (100%),extracellular space (100%);cytosol (100%);plasma membrane (100%),GO:0016936 (100%);GO:0030246 (100%),galactoside binding (100%);carbohydrate binding (100%),GO:0007155 (100%),cell adhesion (100%),IPR001079 (100%);IPR013320 (100%);IPR015533 (100%),Galectin, carbohydrate recognition domain (100%);Concanavalin A-like lectin/glucanase domain superfamily (100%);Galectin-4/6 (100%)
GTVDEFSGAELVDK,Homo sapiens,Eukaryota,Metazoa,,,Chordata,Craniata,Sarcopterygii,Mammalia,,,Euarchontoglires,Primates,Haplorrhini,Simiiformes,Catarrhini,Hominoidea,Hominidae,Homininae,,,Homo,,,,Homo sapiens,,,,3.4.11.9 (200%),Xaa-Pro aminopeptidase (200%),GO:0005576 (100%);GO:0005737 (100%);GO:0005886 (100%),extracellular region (100%);cytoplasm (100%);plasma membrane (100%),GO:0004177 (100%);GO:0046872 (100%);GO:0070006 (100%),aminopeptidase activity (100%);metal ion binding (100%);metalloaminopeptidase activity (100%),,,IPR000587 (100%);IPR000994 (100%);IPR001131 (100%),Creatinase, N-terminal (100%);Peptidase M24 (100%);Peptidase M24B, X-Pro dipeptidase/aminopeptidase P, conserved site (100%)
AAGGLALLGTER,Bacteroidetes,Bacteria,,,,Bacteroidetes,,,,,,,,,,,,,,,,,,,,,,,,,,GO:0005737 (83%);GO:0005886 (83%);GO:0005829 (17%),cytoplasm (83%);plasma membrane (83%);cytosol (17%),GO:0005524 (100%);GO:0046872 (67%);GO:0015462 (17%),ATP binding (100%);metal ion binding (67%);ATPase-coupled protein transmembrane transporter activity (17%),GO:0006605 (100%);GO:0017038 (100%);GO:0065002 (100%),protein targeting (100%);protein import (100%);intracellular protein transmembrane transport (100%),IPR000185 (100%);IPR011115 (100%);IPR011116 (100%),Protein translocase subunit SecA (100%);SecA DEAD-like, N-terminal (100%);SecA Wing/Scaffold (100%)
LLDQGEAGDNVGLLLR,Bacteria,Bacteria,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,GO:0005737 (100%),cytoplasm (100%),GO:0003746 (100%);GO:0003924 (100%);GO:0005525 (100%),translation elongation factor activity (100%);GTPase activity (100%);GTP binding (100%),GO:0006414 (11%),translational elongation (11%),IPR000795 (100%);IPR004160 (100%);IPR004161 (100%),Transcription factor, GTP-binding domain (100%);Translation elongation factor EFTu/EF1A, C-terminal (100%);Translation elongation factor EFTu-like, domain 2 (100%)
LLDLGVLVGSGYHVNPK,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
VYFLNFKPESSDEWK,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
LGLVAVSR,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
AVGFGGDFDGVPR,Homo sapiens,Eukaryota,Metazoa,,,Chordata,Craniata,Sarcopterygii,Mammalia,,,Euarchontoglires,Primates,Haplorrhini,Simiiformes,Catarrhini,Hominoidea,Hominidae,Homininae,,,Homo,,,,Homo sapiens,,,,3.4.13.19 (100%),Membrane dipeptidase (100%),GO:0005615 (100%);GO:0005634 (100%);GO:0005886 (100%),extracellular space (100%);nucleus (100%);plasma membrane (100%),GO:0008235 (100%);GO:0008239 (100%);GO:0008270 (100%),metalloexopeptidase activity (100%);dipeptidyl-peptidase activity (100%);zinc ion binding (100%),GO:0006691 (100%);GO:0006749 (100%);GO:0006805 (100%),leukotriene metabolic process (100%);glutathione metabolic process (100%);xenobiotic metabolic process (100%),IPR000180 (100%);IPR008257 (100%);IPR028536 (100%),Membrane dipeptidase, active site (100%);Peptidase M19 (100%);Dipeptidase 1 (100%)"
`;
